pandas/tests/test_strings.py

```diff
@@ -558,15 +551,15 @@ def test_extract(self):
 
             # single group renames series/index properly
             s_or_idx = klass(['A1', 'A2'])
-            result = s_or_idx.str.extract(r'(?P<uno>A)\d')
-            tm.assert_equal(result.name, 'uno')
-            tm.assert_numpy_array_equal(result, klass(['A', 'A']))
+            result_df = s_or_idx.str.extract(r'(?P<uno>A)\d')
+            result_series = result_df['uno']
+            tm.assert_numpy_array_equal(result_series, klass(['A', 'A']))
 
         s = Series(['A1', 'B2', 'C3'])
         # one group, no matches
         result = s.str.extract('(_)')
-        exp = Series([NA, NA, NA], dtype=object)
-        tm.assert_series_equal(result, exp)
+        exp = DataFrame([NA, NA, NA], dtype=object)
+        tm.assert_frame_equal(result, exp)
```

Comment:
Actually the resultant column name(s) should come from the capture group name(s). I added a test to make sure that extract gives the same output for an input series with and without a name.